crawling bomb attack now uses self-d explosion

KyleAnthonyShepherd opened the pull request:
PR to make crawling bombs use self-d explosion on attack/proximity trigger instead of their death explosion. 

1. Uses `get KILL_UNIT(get MY_ID,TRUE,FALSE);` in the BOS/COB script to detonate the unit.
2. Proximity range reduced to 32, so the crawling bomb has to be hugging the enemy to detonate. Tested vs large units like behemoths, and the short range proxy still works. 
3. Use cylinder targeting so the proxy trigger still works vs ships that may be floating above the crawling bomb.
4. Change proxy trigger to target `NOTAIR`, so the proxy trigger works on submarines.

Isy232 commented:
![temp7](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/41198990/7a7cb7d9-defc-41b3-be8f-af66d396ca51)

WatchTheFort commented:
> Proximity range reduced to 32

Why was this changed? How does this affect balance?

WatchTheFort commented on scripts/Units/armvader.bos:
```diff
@@ -98,7 +98,8 @@ AimWeapon2()
 {
-	emit-sfx 4097 from leg1; //Weapon2 detonates the crawling bomb once weapon1 fires
+	get KILL_UNIT(get MY_ID,TRUE,FALSE);
+	//emit-sfx 4097 from leg1; //Weapon2 detonates the crawling bomb once weapon1 fires
```
Is this line supposed to be uncommented in the scripts? If not, it should be removed.

KyleAnthonyShepherd replied:
Cleaned up

KyleAnthonyShepherd commented:
Cleaned up the unitscripts.
Removed the `emit-sfx 4096 + 1 from base;`, that emitted a self-damaging explosion. Part of the rube goldberg mechanism that was used to detonate crawling bombs.

KyleAnthonyShepherd commented:
> > Proximity range reduced to 32
> 
> Why was this changed? How does this affect balance?

Part of the reasoning behind this PR is to remove the micro tax for crawling bombs, so they have the expected "greedy" behavior (get as close as possible, then use your strongest explosion) when told to attack a target. 
Reducing the proxy range means they will not detonate prematurely when given an attack command. So there is room to micro crawling bombs (self-d early if you expect to be shot and killed soon, even when they are on fire-at-will).

But I have no strong opinions here.

WatchTheFort commented:
@Zecrus2 If you're okay with this, can you merge it? And are there any other kamikaze type units other than crawling bombs that need this change?

This change has been long, long overdue.

Zecrus2 commented:
Before I merge this I'll need to rebalance crawling bombs so the self-d explosion is a bit weaker, as this will end up being a large buff due to ease of use.  I'll try to figure those values out soon.

Isy232 commented:
> Before I merge this I'll need to rebalance crawling bombs so the self-d explosion is a bit weaker, as this will end up being a large buff due to ease of use. I'll try to figure those values out soon.

I agree its a large buff to easy of use, however detonating on contact with the first enemy is still not the optimal use case.
The optimal use case involves setting the crawling bomb to hold fire to run them as far into the middle of the enemy formation as possible.

Are you sure they'd need a nerf to their self-d explosion?
This change would make their normal use case better, however still far from their optimal use case.

WatchTheFort commented:
@Zecrus2 What is the status of this?

WatchTheFort commented:
@Zecrus2 Please give a final review, and add to season two patch notes.

WatchTheFort commented:
Should this still be merged, because it looks like a cleaner implementation?
